README.md

```diff
@@ -2,14 +2,29 @@
 
 - List 3 reasons why asymptotic analysis may be misleading with respect to
   actual performance in practice.
+  
+  1. The assumption of inputs normally being large: Asymptotic analysis typically assumes that input is large often approaching infinity. However, real world problems can typically involve finite input sizes, and some times those inputs are not large enough for asymptotic analysis to be fully realized. For instance, an algorithm that has a time complexity of O(n^2) may have similar performance when compared to O(n log n) when the input is small. Large inputs refering to values in the thousands or millions while small inputs refering to relatively small like 100 elements.
+
+  2. Hidden Constants/ Lower-Order terms: Typically Asymptotic analysis ignores these which can play an important part in actual peformance causing it to be slower.
+
+  3. Hardware factors: Factors like CPU cache behavior, memory access patterns, and paralleism can affect performance of an algorithm, but Asymptotic analysis does not account for them. Causing problems within predictions and performance.
 
 - Suppose finding a particular element in a binary search tree with 1,000
   elements takes 5 seconds. Given what you know about the asymptotic complexity
   of search in a binary search tree, how long would you guess finding the same
   element in a search tree with 10,000 elements takes? Explain your reasoning.
 
+  My first thought is that the program takes 5 seconds to find an element within 1,000 elements, so if it is now 10,000 elements it should take 50 seconds. However, that seemed impractical so I looked at it again with logarithmic and determined it to be around 6 - 7 seconds. So the runtime should be around 7 seconds for a tree with 10,000 assuming the tree is balanced and no other limiing factors. The typical complexity of a balanced binary search tree is O(log n).
```

Comment:
If binary search has logarithmic complexity, why did you start by assuming linear complexity?

Reply:
Honestly I did not mean to and I did not realize I was doing it until I already did, which is why it seemed impractical so I looked at it again with logarithmic complexity.  Do you want me to take the first part out?